ignite/contrib/metrics/regression/r2_score.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import division
+
+import torch
+
+from ignite.contrib.metrics.regression._base import _BaseRegressionEpoch
+
+
+def r2_score_compute_fn(y_pred, y):
+    e = torch.sum((y - y_pred) ** 2) / torch.sum((y - y.mean()) ** 2)
+    return 1 - e.item()
+
+
+class R2Score(_BaseRegressionEpoch):
```

Comment:
@anmolsjoshi yes we can do this while saving the complete history, however it would be better to compute with running parts:
```
SS_tot = \sum_i y_i^2 - (2*n - 1) * y_mean^2
```

Reply:
@vfdev-5 could you clarify the SS_tot function?

Reply:
Actually, I took as a reference this wiki article: https://en.wikipedia.org/wiki/Coefficient_of_determination
or we can take this : https://scikit-learn.org/stable/modules/model_evaluation.html#r2-score-the-coefficient-of-determination

Reply:
Numerator can computed with an accumulative method. Denominator should be expanded and represented as something like in above message. Such approach will avoid saving whole history. What do you think about ?

Reply:
so we keep a running total of y and number of samples, and keep calculating the mean?

Reply:
Fixed implementation